

License header for src/GLX/glvnd_genentry.c:

/*
 * Copyright (c) 2015, NVIDIA CORPORATION.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and/or associated documentation files (the
 * "Materials"), to deal in the Materials without restriction, including
 * without limitation the rights to use, copy, modify, merge, publish,
 * distribute, sublicense, and/or sell copies of the Materials, and to
 * permit persons to whom the Materials are furnished to do so, subject to
 * the following conditions:
 *
 * The above copyright notice and this permission notice shall be included
 * unaltered in all copies or substantial portions of the Materials.
 * Any additions, deletions, or changes to the original source files
 * must be clearly indicated in accompanying documentation.
 *
 * If only executable code is distributed, then the accompanying
 * documentation must state that "this software is based in part on the
 * work of the Khronos Group."
 *
 * THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
 * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
 * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
 * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
 * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
 * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
 */

This was recommended in issue #220.  Mark it so that Clang CFI (Control Flow Integrity) works.

--- a/src/GLX/glvnd_genentry.c.orig	2021-08-24 12:53:07.000000000 -0700
+++ b/src/GLX/glvnd_genentry.c	2021-12-19 23:58:25.632541026 -0800
@@ -45,6 +45,7 @@
 #define GLX_STUBS_COUNT
 #include "g_glx_dispatch_stub_list.h"
 
+__attribute__((used))
 static GLVNDentrypointStub entrypointFunctions[GENERATED_ENTRYPOINT_MAX];
 static char *entrypointNames[GENERATED_ENTRYPOINT_MAX] = {};
 static int entrypointCount = 0;
